clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -106,7 +127,62 @@ public void onFailure(RuntimeException e) {
         return fetchRequestMap.size();
     }
 
+    protected void maybeCloseFetchSessions(final Timer timer) {
+        final List<RequestFuture<ClientResponse>> requestFutures = new ArrayList<>();
+        Map<Node, FetchSessionHandler.FetchRequestData> fetchRequestMap = prepareCloseFetchSessionRequests();
+
+        for (Map.Entry<Node, FetchSessionHandler.FetchRequestData> entry : fetchRequestMap.entrySet()) {
+            final Node fetchTarget = entry.getKey();
+            final FetchSessionHandler.FetchRequestData data = entry.getValue();
+            final FetchRequest.Builder request = createFetchRequest(fetchTarget, data);
+            final RequestFuture<ClientResponse> responseFuture = client.send(fetchTarget, request);
```

Comment:
since we also need to send offsetCommit upon closing, maybe it is best to poll the networkClientDelegate upon closing.

Reply:
Can you look at the new `DefaultBackgroundThread.runAtClose()` method I added? Do we need to update the `CommitRequestManager` to implement the `pollOnClose()` API I added?

Reply:
Thanks @kirktrue - I think we will need to do that.  I created KAFKA-15548 to handle the closing task.

Reply:
I'm hoping to split this into a separate Jira, if possible.